hbase-common/src/main/java/org/apache/hadoop/hbase/HConstants.java

```diff
@@ -181,7 +181,11 @@
   /** Configuration key for the list of master host:ports **/
   public static final String MASTER_ADDRS_KEY = "hbase.masters";
 
-  public static final String MASTER_ADDRS_DEFAULT =  "localhost:" + DEFAULT_MASTER_PORT;
+  // key to the config parameter of server hostname
+  // the specification of server hostname is optional. The hostname should be resolvable from
+  // both master and region server
+  public static final String RS_HOSTNAME_KEY = "hbase.regionserver.hostname";
```

Comment:
Don't add these here. In [the book](http://hbase.apache.org/book.html#hbase_default_configurations), they're considered "expert" configuration params (and the `master` version appears to not be documented at all).

Instead, add them to the `DNS` class, from which they'll be used. Also, it's `InterfaceAudience.Private` so we have more maintenance/refactoring flexibility there. And don't lose their `InterfaceAudience.LimitedPrivate` annotations when you move them.